indico/web/client/js/custom_elements/ind_share_widget.js

```diff
@@ -321,7 +321,7 @@ function ShareWidget({
 }) {
   const [isPopupOpen, setIsPopupOpen] = useState(false);
   const [isMastodonOpen, setMastodonOpen] = useState(false);
-  const shareText = `${eventName} (${eventStartDt}) · Indico (${eventUrl})`;
+  const shareText = `${encodeURIComponent(eventName)} (${eventStartDt}) · Indico (${eventUrl})`;
```

Comment:
To me it feels a bit safer to use `encodeURIComponent` in the place where we actually construct the URL. For example:

```jsx
<Button
  as="a"
  href={`https://twitter.com/intent/tweet?text=${encodeURIComponent(shareText)}`}
  target="_blank"
  basic
  color="blue"
  styleName="share-button"
>
```
Otherwise, it's a bit hard to tell if what you're passing to the URL is encoded or not. What do you think?

Reply:
If we do want to encode the entire share text, then its easier to encapsulate it from where we define it imo:

```diff
diff --git a/indico/web/client/js/custom_elements/ind_share_widget.js b/indico/web/client/js/custom_elements/ind_share_widget.js
index 089dc9ec54..1a1b79a473 100644
--- a/indico/web/client/js/custom_elements/ind_share_widget.js
+++ b/indico/web/client/js/custom_elements/ind_share_widget.js
@@ -321,7 +321,7 @@ function ShareWidget({
 }) {
   const [isPopupOpen, setIsPopupOpen] = useState(false);
   const [isMastodonOpen, setMastodonOpen] = useState(false);
-  const shareText = `${encodeURIComponent(eventName)} (${eventStartDt}) · Indico (${eventUrl})`;
+  const shareText = encodeURIComponent(`${eventName} (${eventStartDt}) · Indico (${eventUrl})`);
   return (
     <Popup
       trigger={
```

Reply:
I agree w/ @tomasr8's comment that it's cleaner to do it when it's actually used. Not feeling super strong about it but since I had to checkout the branch anyway to fix a merge conflict in the changelog, I updated this as well...